### PR TITLE
ES6 ALL THE THINGS.

### PR DESCRIPTION
**Pull request by chuckharmston**
r? @ngokevin 

mo6qn

### CODE REVIEW

File: src/js/components/header.js [hunk 1]
```diff
@@ -45,7 +41,9 @@ let HeaderLogin = React.createClass({
       </div>
     }
   }
-});
-
-
-export default Header;
+}
+HeaderLogin.propTypes = {
```
Comment (ngokevin) [inline src/js/components/header.js]: can this go in the constructor

File: src/css/base.styl [hunk 2]
```diff
@@ -22,7 +22,7 @@ body,
   flex-direction column
 
 main
-  flex 1
+  // flex 1
```
Comment (ngokevin) [inline src/css/base.styl]: ?

File: src/js/actions/submission.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ import urlJoin from 'url-join';
 
 export default class SubmissionActions extends Actions {
   async submitUrl(url) {
+    console.log(url);
```
Comment (ngokevin) [inline src/js/actions/submission.js]: here

File: src/js/components/header.js (hunk 1 above)
Comment (chuckharmston) [inline src/js/components/header.js]: Nope. http://www.newmediacampaigns.com/blog/refactoring-react-components-to-es6-classes

Comment (ngokevin) [inline src/js/components/header.js]: or part of the class as a prop

File: src/css/base.styl (hunk 2 above)
Comment (chuckharmston) [inline src/css/base.styl]: This was causing the footer to overlay the form in weird ways.

Comment (ngokevin) [inline src/css/base.styl]: fixed the body/main/.app heights

**Comment by ngokevin** [comment]
sorry, serious merge conflicts :(

is there a benefit of using ```extends React.Component``` over React.createClass? i like the es6 part of it, but kind of sucks we have to define properties outside of the class definition.

**Comment by ngokevin** [comment]
ES6 + React has a bit of a drawback https://github.com/facebook/react/issues/3040

event handlers aren't  automatically binded with the component instance...

**Comment by ngokevin** [comment]
actually that's more of an es6. not too hard to bind. just need to wrap in a func

**Comment by ngokevin** [comment]
I'm on-board with this. Most components have been ported to ES7 in: https://github.com/mozilla/marketplace-submission/pull/8

**Pull request closed without merging**